// @react95/core/RadioButton/RadioButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _radioButtonStates = require("./radioButtonStates");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Icon = _styledComponents.default.span.withConfig({
  displayName: "RadioButton__Icon",
  componentId: "b8l4i0-0"
})(["width:12px;height:12px;content:'';display:inline-block;position:absolute;left:0;top:0;background-image:url(", ");"], _radioButtonStates.radioUnchecked);

var Text = _styledComponents.default.span.withConfig({
  displayName: "RadioButton__Text",
  componentId: "b8l4i0-1"
})(["padding:1px;user-select:none;position:absolute;top:0;left:18px;"]);

var Field = _styledComponents.default.input.attrs({
  type: 'radio'
}).withConfig({
  displayName: "RadioButton__Field",
  componentId: "b8l4i0-2"
})(["margin:0;opacity:0;&:focus ~ ", ",&:active + ", "{border:1px dotted;padding:0;}&:checked + ", "{background-image:url(", ");}&:disabled + ", "{background-image:url(", ");}&:checked &:disabled + ", "{background-image:url(", ");}"], Text, Text, Icon, _radioButtonStates.radioChecked, Icon, _radioButtonStates.radioUncheckedDisabled, Icon, _radioButtonStates.radioCheckedDisabled);

var Label = _styledComponents.default.label.withConfig({
  displayName: "RadioButton__Label",
  componentId: "b8l4i0-3"
})(["position:relative;margin-bottom:10px;", ""], function (props) {
  return props.disabled && "\n    color: #868686;\n    text-shadow: 0.5px 0.5px #d2d2d2;\n  ";
});

var RadioButton = function RadioButton(_ref) {
  var children = _ref.children,
      onChange = _ref.onChange,
      disabled = _ref.disabled,
      props = _objectWithoutProperties(_ref, ["children", "onChange", "disabled"]);

  return _react.default.createElement(Label, {
    disabled: disabled
  }, _react.default.createElement(Field, _extends({
    onChange: onChange,
    disabled: disabled
  }, props)), _react.default.createElement(Icon, null), _react.default.createElement(Text, null, children));
};

RadioButton.propTypes = {
  children: _propTypes.default.string,
  onChange: _propTypes.default.func,
  disabled: _propTypes.default.bool
};
RadioButton.defaultProps = {
  children: '',
  onChange: function onChange() {},
  disabled: false
};
var _default = RadioButton;
exports.default = _default;
