// @react95/core/Checkbox/Checkbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _check = _interopRequireDefault(require("../shared/assets/pattern/check.svg"));

var _checkDisabled = _interopRequireDefault(require("../shared/assets/pattern/check-disabled.svg"));

var _dropdown = _interopRequireDefault(require("../shared/assets/pattern/dropdown.png"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Icon = _styledComponents.default.span.withConfig({
  displayName: "Checkbox__Icon",
  componentId: "yxi6zd-0"
})(["width:12px;height:12px;content:'';display:inline-block;position:absolute;left:0;border-left:1px solid #868a8e;border-top:1px solid #868a8e;box-shadow:inset -1px -1px 0 0 #c3c7cb,inset 1px 1px 0 0 #000000,0.5px 0.5px 0 0.5px #ffffff;background-color:white;background-repeat:no-repeat;background-position:center center;background-size:7px 7px;"]);

var Text = _styledComponents.default.span.withConfig({
  displayName: "Checkbox__Text",
  componentId: "yxi6zd-1"
})(["padding:1px;user-select:none;"]);

var Field = _styledComponents.default.input.withConfig({
  displayName: "Checkbox__Field",
  componentId: "yxi6zd-2"
})(["width:12px;height:12px;margin:0;position:absolute;top:0;left:0;opacity:0;&:focus ~ ", ",&:active ~ ", "{border:1px dotted;padding:0;}&:checked + ", "{background-image:url('", "');}&:checked &:disabled + ", "{background-image:url('", "'),url('", "');background-size:7px 7px,1.9px 1.9px;}&:disabled + ", "{background-color:#c0c0c0;}"], Text, Text, Icon, _check.default, Icon, _checkDisabled.default, _dropdown.default, Icon);

Field.displayName = 'Field';

var Label = _styledComponents.default.label.withConfig({
  displayName: "Checkbox__Label",
  componentId: "yxi6zd-3"
})(["display:inline-block;height:15px;line-height:1.5;position:relative;margin:4px 0;padding-left:20px;", ";"], function (_ref) {
  var disabled = _ref.disabled;
  return disabled && "\n    color: #868686;\n    text-shadow: 0.5px 0.5px #d2d2d2;\n  ";
});

var Checkbox = function Checkbox(_ref2) {
  var children = _ref2.children,
      label = _ref2.label,
      value = _ref2.value,
      style = _ref2.style,
      checked = _ref2.checked,
      disabled = _ref2.disabled,
      onChange = _ref2.onChange,
      rest = _objectWithoutProperties(_ref2, ["children", "label", "value", "style", "checked", "disabled", "onChange"]);

  return _react.default.createElement(Label, _extends({}, style, {
    checked: checked,
    disabled: disabled
  }), _react.default.createElement(Field, _extends({
    type: "checkbox",
    value: value,
    onChange: onChange,
    checked: checked,
    disabled: disabled
  }, rest)), _react.default.createElement(Icon, null), _react.default.createElement(Text, null, children || label));
};

Checkbox.propTypes = {
  value: _propTypes.default.string,
  label: _propTypes.default.string,
  children: _propTypes.default.string,
  checked: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  onChange: _propTypes.default.func
};
Checkbox.defaultProps = {
  value: undefined,
  label: 'Checkbox',
  children: '',
  checked: null,
  disabled: null,
  onChange: function onChange() {}
};
var _default = Checkbox;
exports.default = _default;
