// @react95/core/List/ListItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _List = _interopRequireDefault(require("./List"));

var _rightcaret = _interopRequireDefault(require("../shared/assets/pattern/rightcaret.svg"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Item = _styledComponents.default.li.withConfig({
  displayName: "ListItem__Item",
  componentId: "sc-1w7jytv-0"
})(["position:relative;display:flex;align-items:center;margin:0;padding:2px 6px;i{margin-right:10px;}&:hover{background-color:#00007f;color:white;}ul{display:none;position:absolute;top:-2px;left:97%;color:#000;i{width:18px;height:20px;margin-right:6px;}}svg{fill:white;}", ";", ";"], function (_ref) {
  var icon = _ref.icon;
  return !icon && "padding-left: 26px;";
}, function (_ref2) {
  var hasList = _ref2.hasList;
  return hasList && "\n    &:after {\n      position: absolute;\n      content: '';\n      background-color: #000;\n      mask-image: url('".concat(_rightcaret.default, "');\n      mask-position: center center;\n      mask-size: 5px 8px;\n      mask-repeat: no-repeat;\n      width: 5px;\n      height: 8px;\n      right: 8px;\n    }\n\n    &:hover {\n      &:after {\n        background-color: #FFF;\n      }\n\n      ul {\n        display: block;\n      }\n    }\n    ");
});

var ListItem = function ListItem(_ref3) {
  var icon = _ref3.icon,
      children = _ref3.children,
      onClick = _ref3.onClick,
      rest = _objectWithoutProperties(_ref3, ["icon", "children", "onClick"]);

  return _react.default.createElement(Item, _extends({}, rest, {
    onClick: onClick,
    icon: icon,
    hasList: _react.default.Children.map(children, function (children) {
      return children.type === _List.default;
    }).some(function (child) {
      return child;
    })
  }), icon && _react.default.createElement(_Icon.default, {
    name: icon
  }), children);
};

ListItem.propTypes = {
  icon: _propTypes.default.string,
  children: _propTypes.default.node.isRequired,
  onClick: _propTypes.default.func
};
ListItem.defaultProps = {
  icon: '',
  onClick: function onClick() {}
};
var _default = ListItem;
exports.default = _default;
